### DISCUSSION TITLE
Display action buttons

### DISCUSSION THREAD

**Discussion by mkts02** [Power-Components/livewire-powergrid#1141] [Q&A]
hi, im trying to get the action btns to show on the table, any ideas?
thanks

      Public function setUp(): array
    {
        $this->showCheckBox();

        return [
            Responsive::make()
            ->fixedColumns('rut', 'name', Responsive::ACTIONS_COLUMN_NAME),
            Exportable::make('export')
                ->striped()
                ->type(Exportable::TYPE_XLS, Exportable::TYPE_CSV),
            Header::make()->showSearchInput(),
            Footer::make()
                ->showPerPage()
                ->showRecordCount(),
        ];
    }

    public function datasource(): Builder
    {
        return Company::query();
    }

    public function relationSearch(): array
    {
        return [];
    }

    public function addColumns(): PowerGridColumns
    {
        return PowerGrid::columns()
            ->addColumn('rut')

           /** Example of custom column using a closure **/
            ->addColumn('rut_lower', fn (Company $model) => strtolower(e($model->rut)))

            ->addColumn('name')
            ->addColumn('phone')
            ->addColumn('created_at_formatted', fn (Company $model) => Carbon::parse($model->created_at)->format('d/m/Y H:i:s'));
    }

    public function columns(): array
    {
        return [
            Column::make('Rut', 'rut')
                ->sortable()
                ->searchable(),

                Column::make('Name', 'name')
                ->sortable()
                ->searchable(),

            Column::make('Phone', 'phone')
                ->sortable()
                ->searchable(),

            Column::make('Created at', 'created_at_formatted', 'created_at')
                ->sortable(),

        ];
    }

    public function filters(): array
    {
        return [
            Filter::inputText('rut')->operators(['contains']),
            Filter::inputText('name')->operators(['contains']),
            Filter::inputText('phone')->operators(['contains']),
            Filter::inputText('address')->operators(['contains']),
            Filter::datetimepicker('created_at'),
        ];
    }

    #[\Livewire\Attributes\On('edit')]
    public function edit($rowId): void
    {
        $this->js('alert('.$rowId.')');
    }

    public function actions(\App\Models\Company $row): array
    {
        return [
            Button::add('edit')
                ->slot('Edit: '.$row->id)
                ->id()
                ->class('pg-btn-white dark:ring-pg-primary-600 dark:border-pg-primary-600 dark:hover:bg-pg-primary-700 dark:ring-offset-pg-primary-800 dark:text-pg-primary-300 dark:bg-pg-primary-700')
                ->dispatch('edit', ['rowId' => $row->id]),
        ];
    }

    
    public function actionRules($row): array
    {
       return [
            // Hide button edit for ID 1
            Rule::button('edit')
                ->when(fn($row) => $row->id === 1)
                ->hide(),
        ];
    }

**Comment by luanfreitasdev** [comment]
https://github.com/Power-Components/livewire-powergrid/issues/1142

Fixed: https://github.com/Power-Components/livewire-powergrid/pull/1143
